packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -398,10 +384,17 @@ export async function sourceNodes(
         })
       }
     })
+  }
+
+  // allow node to gc if it needs to
+  // @ts-ignore
+  newOrUpdatedEntries = undefined
```

Comment:
You can achieve this without unsetting values with introducing closure / scope - for example like in https://github.com/gatsbyjs/gatsby/blob/68510e191507caabdfd19dfb86c7a595c9fbc0a3/packages/gatsby/src/commands/build.ts#L567-L601

Leaving that scope make variables inside of it no longer strongly retain objects in memory and hence being subject to gc. It also just visually send message better that some variable is only relevant for few next statements and just tidying up scope of the function (including not littering code editors intellisense/code completion with variables that are no longer relevant)

Reply:
Generally I agree but in this case there's already too much nesting in this code and that would reduce readability quite a bit. It's already more difficult to scan than it should be and adding another scope here will make it worse imo. 

If we go that route I think it'd be better to move this logic into a new function (much of the code here would benefit from being in smaller chunks anyway). But I already moved a lot of code and didn't want to make the diff even larger.